### DISCUSSION TITLE
How to change the waveguide width

### DISCUSSION THREAD

**Discussion by nisreenalnaji** [SiEPIC/SiEPIC_EBeam_PDK#257] [General]
Hey, What is the proper way to change the width of a waveguide in K-layout?

**Comment by jevillegasd** [comment]
Hey,

You can change the waveguide width by updating the WAVEGUIDE.xml specification file. This is where all the parameters for your waveguide are stored. Do note that the CML in the model in Interconnect includes a model that spans [0.3 - 3] um, so unless you use your own model of waveguides, the circuit simulation will only work in this range. 

The waveguide specification file (WAVEGUIDE.xml) is stored in the KLayout/SiEPIC installation folder, you can find it in Windows as:
C:\Users\$CURRENTUSER$\KLayout\salt\siepic_ebeam_pdk\EBeam

The typical waveguide specification will have this format:

```
<waveguide>
    <name>A name for your waveguide </name>
    <CML>EBeam</CML>
    <model>ebeam_wg_integral_1550</model>
    <bezier>0.20</bezier>
    <radius>5.0</radius>
    <width>0.5</width>
    <crossing>
        <crossing_cell>ebeam_crossing4</crossing_cell>
        <crossing_library>EBeam</crossing_library>
        <crossing_offset>(0,0)</crossing_offset>
    </crossing>
    <sbends>1</sbends>
    <component>
        <layer>Si</layer>
        <width>0.5</width>
        <offset>0.0</offset>
    </component>
    <component>
        <layer>DevRec</layer>
        <width>1.5</width>
        <offset>0.0</offset>
    </component>
    <component>
        <layer>Waveguide</layer>
        <width>0.5</width>
        <offset>0.0</offset>
    </component>
</waveguide>
```
Each element in a 'component' tag will define something that gets drawn on the layout, so you can simultaneously draw several layers at a time, each with a different width. In your case, if you need to change the waveguide width you will have to change the value in several places. In this example, you will need to at least change it in line 7  `<width>0.5</width>`, and in the component that draws in the Silicon layer.  The first will change the component specification, so that when your run a simulation the right value gets used, and the second will actually change the width in the layout.

```
<component>
        <layer>Waveguide</layer>
        <width>0.5</width>
        <offset>0.0</offset>
</component>
```

I hope this helps 👍🏽
